src/books.js

```diff
@@ -1,109 +1,189 @@
+    },
+  },
+  {
+    title: "Pride and Prejudice",
+    pages: 279,
+    author: "Jane Austen",
+    details: {
+      language: "English",
+      description: "One of the most popular novels in the English language...",
+    },
+  },
+  {
+    title: "Educated - A Memoir",
+    pages: 352,
+    author: "Tara Westover",
+    details: {
+      language: "English",
+      description:
+        "Educated is an account of the struggle for self-invention...",
+    },
+  },
+  {
+    title: "The Art of Learning",
+    pages: 288,
+    author: "Josh Waitzkin",
+    details: {
+      language: "English",
+      description:
+        "The Art of Learning takes readers through Waitzkin's unique journey to excellence. He explains in clear detail how a well-thought-out, principled approach to learning is what separates success from failure.",
+    },
+  },
+];
 
 // Iteration 2 | Book Details
-function getBookDetails() {
-  // Your code here:
 
+const book = {
+  title: "Educated - A Memoir",
+  pages: 352,
+  author: "Tara Westover",
+  details: {
+    language: "English",
+    description: "Educated is an account of the struggle for self-invention...",
+  },
+};
+function getBookDetails(book) {
+  return `${book.title} - ${book.author} - ${book.pages} pages`;
 }
 
-
-
 // Iteration 3 | Delete Language
 // Your code here:
 
+booksArray.forEach((book) => {
+  delete book.details.language;
+});
 
-
+console.log(booksArray);
 
 // Iteration 4 | Estimated Reading Time
 // Your code here:
 
+booksArray.forEach((book) => {
+  let readingTime = Math.round((book.pages * 500) / 90);
 
+  book.readingTime = readingTime;
 
+  console.log(booksArray);
+});
 
 // Bonus: Iteration 5 | Books Dictionary
 
 /* The `dictionary` is an object containing books grouped by author. 
  The book info is stored in arrays with structure: [title, pages]. 
 */
 const dictionary = {
-    "J. K. Rowling": [
-        ["Harry Potter and the Philosopher's Stone", 223],
-        ["Harry Potter and the Chamber of Secrets", 251],
-        ["Harry Potter and the Prisoner of Azkaban", 317],
-        ["Harry Potter and the Goblet of Fire", 636],
-    ],
-    "Neal Stephenson": [
-        ["Cryptonomicon", 928],
-        ["Anathem", 1008],
-        ["Fall; or, Dodge in Hell", 896],
-    ],
-    "Malcolm Gladwell": [
-        ["Outliers", 320],
-        ["Blink", 287],
-    ],
+  "J. K. Rowling": [
+    ["Harry Potter and the Philosopher's Stone", 223],
+    ["Harry Potter and the Chamber of Secrets", 251],
+    ["Harry Potter and the Prisoner of Azkaban", 317],
+    ["Harry Potter and the Goblet of Fire", 636],
+  ],
+  "Neal Stephenson": [
+    ["Cryptonomicon", 928],
+    ["Anathem", 1008],
+    ["Fall; or, Dodge in Hell", 896],
+  ],
+  "Malcolm Gladwell": [
+    ["Outliers", 320],
+    ["Blink", 287],
+  ],
 };
 
-function booksByAuthor() {
-  // Your code here:
-  
+function booksByAuthor(dictionary) {
+  if (!dictionary || typeof dictionary !== "object") {
+    return []; 
```

Comment:
For a situation where the dictionary is not an object, this wouldn't be the ideal response to this situation. Best to alert the user/dev in some way that the diction is in the wrong format. You can either do a `console.log` before the return, or better yet, thrown an error. We'll cover error handling soon, so this is fine for now. At least it attempts to handle the situation.